components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt

```diff
@@ -209,11 +211,35 @@ class GeckoEngineTest {
             TrackingProtectionPolicy.SAFE_BROWSING_MALWARE,
             TrackingProtectionPolicy.SAFE_BROWSING_PHISHING
         ).categories, contentBlockingSettings.categories)
+        assertEquals(contentBlockingSettings.cookieBehavior, CookiePolicy.ACCEPT_NON_TRACKERS.id)
         assertTrue(engine.settings.testingModeEnabled)
         assertEquals("test-ua", engine.settings.userAgentString)
         assertEquals(PreferredColorScheme.Light, engine.settings.preferredColorScheme)
         assertFalse(engine.settings.allowAutoplayMedia)
         assertTrue(engine.settings.suspendMediaWhenInactive)
+
+        engine.settings.trackingProtectionPolicy =
+            TrackingProtectionPolicy.select(
+                TrackingProtectionPolicy.AD,
+                cookiePolicy = CookiePolicy.ACCEPT_ONLY_FIRST_PARTY
+            )
+
+        assertEquals(CookiePolicy.ACCEPT_ONLY_FIRST_PARTY.id, contentBlockingSettings.cookieBehavior)
+
+        engine.settings.trackingProtectionPolicy = TrackingProtectionPolicy.none()
+
+        assertEquals(CookiePolicy.ACCEPT_ALL.id, contentBlockingSettings.cookieBehavior)
+
+        engine.settings.trackingProtectionPolicy =
```

Comment:
This assertion seems redundant? Isn't it the same as line 221+. You select a TP policy with a cookiePolicy and assert that the cookie policy is set correctly. Seems to me both blocks test the same?

Reply:
Fixed, thanks!